### PR TITLE
Add support for regression tests on NCEP RDHPC orion machine 

### LINKED ISSUE #441: enable matrix_ncep on orion
We should be able to run matrix_ncep on orion.  The first issue was to change mpirun to srun which is the command we should be using on hera too for slurm.  Still debugging issues on orion that include: 
-- I think the parmetis library needs to be rebuilt now that we're using hpc-stack modules on orion  (@aliabdolali can you help with this?) 
-- the oasis tests fail see question in issue #440

This work is being done on: https://github.com/JessicaMeixner-NOAA/WW3/tree/orion

When completed, the hope is to be able to use the hpc-stack modules on orion and run the WW3 regression tests on orion as well as hera.

### LINKED ISSUE #442: Input needs updating
**Describe the bug**
https://github.com/NOAA-EMC/WW3/blob/develop/regtests/ww3_tp2.17/input/ww3_multi_grdset_a1.inp#L9: 
 'inla'  'native' 'native' 'native' 'no' 'no' 'no' 'no'   1  1  0.00 1.00  F
should be: 
 'inla'  'native' 'native' 'native' 'no' 'no' 'no' 'no' 'no' 'no'  1  1  0.00 1.00  F

**To Reproduce**
 Running regression tests with similar options: 
 run_test -b slurm -c orion.intel -S -T -s MPI -s NO_PDLIB -w work_ma1 -m grdset_a1 -f -p srun -n 24 -o all ../model ww3_tp2.17

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Pull Request Summary

Adds support for NCEP's RDHPCS resource orion and updates hera to use "srun" as recommended. 

## Description
This PR allows for running of the regression tests on Orion.  No comparison is done on orion and there are still a few errors due to a NetCDF issue on orion when using partitions (see: https://github.com/NOAA-EMC/WW3/issues/451) 

### Issue(s) addressed
* Fixes #441

### Check list  
* Is your feature branch up to date with the authoritative repository (NOAA/develop)? yes
* Make sure you have checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop) and [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) - no version number update 
* Please list appropriate labels code managers should add for this PR:   
 _bug_, _documentation_, _enhancement_, _new feature_, ..
 
* Reviewers: @aliabdolali 

### Commit Message
* Update NCEP regression tests to use srun and enable orion 

### Testing
* How were these changes tested? WW3 standalone regtests
* Are the changes covered by regression tests? yes
* Have regression tests been run? yes
* Which compiler / HPC you used to run the regression tests in the PR?  hera.intel and orion.intel 
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):    
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/7130957/matrixDiff.txt)
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/7130958/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/7130960/matrixCompSummary.txt)
Please indicate the expected changes in the outputs ([excluding the known list of non-identical tests](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-master#4-look-at-results)). Just the typical non-b4b tests: 

```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (9 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (8 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (10 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (9 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_07/./work_PR3_UQ                     (3 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (6 files differ)
ww3_tp2.14/./work_OASACM4                     (0 files differ)
ww3_tp2.14/./work_OASICM                     (0 files differ)
ww3_tp2.14/./work_OASACM6                     (0 files differ)
ww3_tp2.14/./work_OASOCM                     (0 files differ)
ww3_tp2.14/./work_OASACM2                     (0 files differ)
ww3_tp2.14/./work_OASACM5                     (0 files differ)
ww3_tp2.14/./work_OASACM                     (0 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.17/./work_ma1                     (0 files differ)
ww3_tp2.17/./work_mc1                     (1 files differ)
ww3_ufs1.1/./work_b                     (0 files differ)
ww3_ufs1.2/./work_b                     (0 files differ)
ww3_ufs1.3/./work_a                     (1 files differ)

```

* Please list which labels code managers should add to indicate code changes:    
none

### CODE REVIEW

File: model/bin/cmplr.env [hunk 1]
```diff
@@ -136,7 +136,7 @@ if [ "$cmplr" == "intel" ]          || [ "$cmplr" == "intel_debug" ]    || [ "$c
   # OPTIONS - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
   # common options
-  optc='-c -module $path_m -no-fma -ip -g -i4 -real-size 32 -fp-model precise -assume byterecl -convert big_endian -fno-alias -fno-fnalias -sox'
+  optc='-c -module $path_m -no-fma -ip -g -traceback -i4 -real-size 32 -fp-model precise -assume byterecl -convert big_endian -fno-alias -fno-fnalias -sox'
```
Comment (aliabdolali) [inline model/bin/cmplr.env:139]: Does traceback change the run time?

File: regtests/bin/matrix_ncep [hunk 2]
```diff
@@ -155,7 +157,7 @@
 
   if [ "$batchq" = 'slurm' ]
   then
-    export  mpi='mpirun'
+    export  mpi='srun'
```
Comment (aliabdolali) [inline regtests/bin/matrix_ncep:159]: Hera scheduler is also slurm, does it work on Hera?
Do you want to test the matrix on Orion only or Hera and Orion?

File: model/bin/cmplr.env (hunk 1 above)
Comment (JessicaMeixner-NOAA) [reply model/bin/cmplr.env:139]: traceback should not effect runtime and will provide better error information

File: regtests/bin/matrix_ncep (hunk 2 above)
Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_ncep:159]: This has been tested on both hera and orion.  The matrix can be tested on both.  On orion, there are still a few remaining netcdf issues that have not been resolved.  On hera, when submitted the matrix reproduced the existing develop matrix output.  I would expect the same now too.

Comment (aliabdolali) [reply regtests/bin/matrix_ncep:159]: thanks, I will test it on one machine then today.

Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_ncep:159]: This needs to be tested on hera for checking matrix.comp and the new capability this is adding should then be tested on orion (as we cannot test that it works on orion on hera).

Comment (aliabdolali) [reply regtests/bin/matrix_ncep:159]: good point. I'll do it today as well.

**Comment by aliabdolali** [comment]
tests went well on Orion except the ones related to NetCDF libraries on orion when using partitions (see: #451)

**Comment by JessicaMeixner-NOAA** [comment]
If the ww3_tp2.16/./work_MPI_OMPH test is run in develop and in this branch from a fresh clone and only that test is run, then the output is identical (same thing for running 2 times from this branch).  However, when you run multiple tests in this branch, we do not get reproducible results or reproduce the develop branch.  Typically this type of error is because w3_new has an error.  I'm continuing to investigate, but it seems like an unrelated error to this branch but for whatever reason we see it with srun (which we should be using) and not when using mpirun.

**Comment by JessicaMeixner-NOAA** [comment]
Another test I have tried is to use the tp2.10 OMPH switch file for the tp2.16 test since it's immediately following it, which did not help either.  However, the other thing is tp2.10 is a known not b4b tests (https://github.com/NOAA-EMC/WW3/issues/321) and tp2.16 is very similar, just a different grid (arctic), so perhaps we were just lucky before that we were getting reproducibility there?     

@aliabdolali I've looked fairly extensively at this point.  I'm not sure why srun hasn't been used all along and since every other test is the same except this one which is very similar to a known-not b4b test, I think our options at this point are
 1. wait until the b4b issues are solved, which no one is working on currently and will keep this PR on hold for a long time or 
 2. decide we should move forward with this PR as is knowing it will add another test to the EMC not-b4b list and add a note to issue #321 to include the tp2.16 OMPH test

**Comment by aliabdolali** [comment]
@JessicaMeixner-NOAA I think we should move forward, srun is more efficient especially with OMP options. So, given that, I will rerun the tests and will merge afterward.
Thanks for all the detailed diagnoses.

**Comment by aliabdolali** [comment]
The tests ran successfully with known non-b4b tests _ ww3_tp2.16

**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (9 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (8 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (8 files differ)
mww3_test_07/./work_PR3_UQ                     (3 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_ufs1.3/./work_a                     (1 files differ)

test ww3_tp2.16/./work_MPI_OMPH  is added to the list of non-identical cases with a note in #321

**Review by aliabdolali — APPROVED** [review]

**Pull request merged**